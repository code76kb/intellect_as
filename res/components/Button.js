import React from "react";
import { Text, StyleSheet, TouchableHighlight } from "react-native";
import PropTypes from 'prop-types'
export default function Button(props) {

    let coolDown=true;
    function onClickWithThrottle(){
        // console.log("Button CoolDown :"+coolDown);
        if(props.disabled)
            return;
        if(coolDown){
            props.onPress();
            coolDown = false;
        }
        else{
            setTimeout(()=>coolDown=true,props.throttleTime);
        }
    }

    return (
        <TouchableHighlight
            style={props.buttonStyle}
            activeOpacity={0.6}
            underlayColor="#DDDDDD"
            onPress={onClickWithThrottle}>
            <Text style={props.labelStyle}>{props.label}</Text>
        </TouchableHighlight>
    );
}


const styles = StyleSheet.create({
    button: {
        width:100,
        height: 40,
        backgroundColor: 'ivory',
        borderRadius: 5,
        justifyContent: 'center',
        alignItems: 'center',
        marginVertical: 15,
        paddingHorizontal:20
    },
    label: {
        color: 'black',
        fontSize: 16,
    },
});

Button.propTypes = {
    onPress: PropTypes.func.isRequired,
    label: PropTypes.string,
    buttonStyle: PropTypes.any,
    labelStyle: PropTypes.any,
    throttleTime: PropTypes.number,
    disabled: PropTypes.bool,
}

Button.defaultProps = {
    buttonStyle: styles.button,
    labelStyle: styles.label,
    throttleTime: 300,
    disabled:false,
}
