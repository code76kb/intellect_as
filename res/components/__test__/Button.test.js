import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../Button';

it('Render Button Com :',()=>{
    const tree = renderer.create(
        <Button
        onPress={()=>{}}
        label="NEXT"
        buttonStyle={{
          width: '90%',
          height: 40,
          borderRadius: 20,
          backgroundColor: 'white',
          justifyContent: 'center',
          alignItems: 'center',
          alignSelf: 'center'
        }}
      />   ).toJSON();
      expect(tree).toMatchSnapshot();
})

