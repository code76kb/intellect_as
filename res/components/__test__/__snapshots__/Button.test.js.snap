// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Button Com : 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "white",
      "borderRadius": 20,
      "height": 40,
      "justifyContent": "center",
      "width": "90%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 16,
      }
    }
  >
    NEXT
  </Text>
</View>
`;
